### PR TITLE
[kernel] rtthread.h 增加#ifdef RT_USING_HEAP 预编译

### PR DESCRIPTION
**Pull request by mysterywolf**
## 拉取/合并请求描述：(PR description)

[

]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in the submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use a web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality
- [x] 本拉取/合并符合[RT-Thread代码规范](../documentation/coding_style_cn.md) This PR complies with [RT-Thread code specification](../documentation/coding_style_en.txt)

### CODE REVIEW

File: include/rtthread.h [hunk 1]
```diff
@@ -128,24 +132,26 @@ rt_err_t rt_thread_init(struct rt_thread *thread,
...
-rt_err_t rt_thread_delete(rt_thread_t thread);
-
 rt_err_t rt_thread_yield(void);
 rt_err_t rt_thread_delay(rt_tick_t tick);
 rt_err_t rt_thread_delay_until(rt_tick_t *tick, rt_tick_t inc_tick);
 rt_err_t rt_thread_mdelay(rt_int32_t ms);
 rt_err_t rt_thread_control(rt_thread_t thread, int cmd, void *arg);
 rt_err_t rt_thread_suspend(rt_thread_t thread);
 rt_err_t rt_thread_resume(rt_thread_t thread);
+void rt_thread_exit(void);
```
Comment (BernardXiong) [inline include/rtthread.h:154]: 这个api没必要露出

File: include/rtthread.h [hunk 2]
```diff
@@ -48,9 +48,11 @@ void rt_object_init(struct rt_object         *object,
                     enum rt_object_class_type type,
                     const char               *name);
 void rt_object_detach(rt_object_t object);
+#ifdef RT_USING_HEAP
```
Comment (BernardXiong) [inline include/rtthread.h:51]: 这些更多的是针对声明，有声明而未有实现，会在最终链接时报错。

**Review by BernardXiong — CHANGES_REQUESTED** [review]

File: include/rtthread.h (hunk 1 above)
Comment (mysterywolf) [reply include/rtthread.h]: revert 回来了

**Comment by BernardXiong** [comment]
先不考虑吧，等到完善注释时，看看是否加入这块的考虑。

**Comment by mysterywolf** [comment]
注释完善的差不多喽

**Comment by Guozhanxin** [comment]
加上之后，看起来更清楚了点。

**Pull request merged**